### ISSUE TITLE
disk plugin test failure on ZFS

### ISSUE THREAD

**Issue by msarahan** [guildai/guildai#328]
I have a ubuntu 21.10 machine that used the ZFS setup with disk encryption at install time. One of the disk plugin tests fails:

```
  disk-plugin: **********************************************************************
File "/home/msarahan/code/guildai/guild/tests/disk-plugin.md", line 16, in disk-plugin.md
Failed example:
    vals = plugin.read_summary_values(0)
Exception raised:
    Traceback (most recent call last):
      File "/home/msarahan/mambaforge/envs/guildai_38/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest disk-plugin.md[3]>", line 1, in <module>
        vals = plugin.read_summary_values(0)
      File "/home/msarahan/code/guildai/guild/plugins/disk.py", line 47, in read_summary_values
        return self._disk_stats()
      File "/home/msarahan/code/guildai/guild/plugins/disk.py", line 53, in _disk_stats
        for dev_name, last, cur in _zip_physical_disk_stats(
      File "/home/msarahan/code/guildai/guild/plugins/disk.py", line 83, in _zip_physical_disk_stats
        assert fullname.startswith('/dev/'), fullname
    AssertionError: rpool/ROOT/ubuntu_3a6epb
**********************************************************************
```

I'll try to work this one out, since recreating the ZFS setup would probably take some time.

**Comment by gar1t** [comment]
Ick, that code that sets `name` should be a function.

I think the assertion there is a bit aggressive. Maybe a set of known rules (e.g. `/dev/...`, `rpool/.../...`, etc.) with a fallback (e.g. `guild.util.safe_filename(fullname)`?

Just my 2 cents.

**Comment by msarahan** [comment]
fixed by #331; closing
